### PR TITLE
[SEMI-MODULAR] mini space ninja rework and fixes

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

Replaces the ninja's murder weapon with a more fun pocket energy tonfa

- Retracts like telebaton, teleports you when retracted
- Breaks bones when used harmfully
- Fits in bags
- makes the ninja "less than lethal" but still very deadly and subversive without outright killing people
- is also a boomerang if that wasn't enough

Open the the idea of the katana as a reward.
Also tries to the the player prefs loading issue. 
dreammaker_IpH29nllyA
dreammaker_4XhEpE3svi

## Changelog

:cl:
balance: replaced the ninja katana with a tonfa
/:cl:

### CODE REVIEW

**Comment by kiwedespars** [comment]
are we going to replace the changeling's armblade with a bone tonfa too

**Comment by zeroisthebiggay** [comment]
'energy tonfa'
>look inside

>`it's made of telebaton`

why is this needed now

**Comment by zeroisthebiggay** [comment]
why does ninja need a tonfa again? all of their objectives don't require killing or even let alone interacting with a living human being besides a cyborg but they aren't human

the only reason a ninja'd be fighting is the inevitable fools trying to duel them `(how are you gonna fight to the death with a fucking tonfa)`
and self defense `(fight lasers with tonfas)`

not only is it a critical breach of ninja's character to replace their iconic sword with a janky and unintuitive tonfa that does no fucking damage while being attached to one of otherwise the most powerful and oppressive to fight antagonists it's just not fluid with their goals.

give them a tonfa they can deploy from their suit to beat up assistants that are chasing them without slicing out all of their internal organs but by god ninja needs a deterrent to be validhunted immediately upon sight in the form of being disemboweled if you fuck with them

**Comment by ReturnToZender** [comment]
Best joke PR so far

**Comment by Serxule** [comment]
ah yes ninjas, with their iconic retractable police batons that fracture bones, i didnt realize it was april first

**Comment by nevimer** [comment]
image
given batons fit the hackerman figure better imo

File: code/modules/events/ghost_role/space_ninja.dm [hunk 1]
```diff
@@ -43,7 +43,10 @@
 		ninja.dna.update_dna_identity()
 	else
 		ninja.randomize_human_appearance(~(RANDOMIZE_NAME|RANDOMIZE_SPECIES))
-		ninja.dna.update_dna_identity()
+		ninja.dna.update_dna_identity() // BUBBER EDIT: MOVED TO NINJA MODULE
+*/
+	sleep(1 SECONDS) // BUBBER EDIT - Ninja Module
```
Comment (BurgerLUA) [inline code/modules/events/ghost_role/space_ninja.dm:48]: Sleep procs should be avoided as much as possible in favor of callbacks.

**Review by BurgerLUA — APPROVED** [review]
Looks good, but just 1 code standards issue.

**Pull request closed without merging**